rasa/engine/recipes/graph_recipe.py

```diff
@@ -0,0 +1,52 @@
+import logging
+
+from rasa.engine.recipes.recipe import Recipe
+from rasa.engine.graph import GraphModelConfiguration
+from rasa.shared.constants import DOCS_URL_GRAPH_RECIPE
+from rasa.shared.data import TrainingType
+from rasa.shared.utils.common import mark_as_experimental_feature
+from rasa.shared.utils.io import raise_warning
+from rasa.engine.graph import GraphSchema
+from rasa.nlu.classifiers.regex_message_handler import RegexMessageHandler
+
+from typing import Dict, Text, Any
+
+
+logger = logging.getLogger(__name__)
+
+
+class GraphV1Recipe(Recipe):
+    """Recipe which converts the graph model config to train and predict graph."""
+
+    name = "graph.v1"
+
+    def graph_config_for_recipe(
+        self,
+        config: Dict,
+        cli_parameters: Dict[Text, Any],
+        training_type: TrainingType = TrainingType.BOTH,
+        is_finetuning: bool = False,
+    ) -> GraphModelConfiguration:
+        """Converts the default config to graphs (see interface for full docstring)."""
+        mark_as_experimental_feature("graph recipe")
+        if cli_parameters or is_finetuning:
+            raise_warning(
+                "Unlike the Default Recipe, Graph Recipe does not utilize CLI "
+                "parameters or finetuning and these configurations will be ignored. "
+                "Add configuration to the recipe itself if you want them to be used.",
+                docs=DOCS_URL_GRAPH_RECIPE,
+            )
+        # TODO: Add core_target and nlu_target to graph configuration options so we
```

Comment:
hmm this is interesting - I hadn't considered this...
I reckon this is worth doing now.

Reply:
I wasn't sure if this would be of much use so I had left as a comment but should not be hard to change this. It would provide even more flexibility for researchers.